### PR TITLE
ipnstate, ipnlocal: add AuthURL to status

### PR DESCRIPTION
**Pull request by crawshaw**

### CODE REVIEW

File: ipn/ipnstate/ipnstate.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ import (
 // Status represents the entire state of the IPN network.
 type Status struct {
 	BackendState string
+	AuthURL      string
```
Comment (bradfitz) [inline ipn/ipnstate/ipnstate.go:30]: these fields could all use some docs, but this one in particular is a bit subtle in its meaning: does non-empty only mean that we're not logged in? For which BackendState is this populated?

File: ipn/ipnlocal/local.go [hunk 2]
```diff
@@ -213,6 +213,7 @@ func (b *LocalBackend) UpdateStatus(sb *ipnstate.StatusBuilder) {
 	defer b.mu.Unlock()
 
 	sb.SetBackendState(b.state.String())
+	sb.SetAuthURL(b.authURL)
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:216]: out of paranoia, could you only set this if `b.state` is waiting-for-login, and set it to `""` in all other cases?

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnstate/ipnstate.go (hunk 1 above)
Comment (crawshaw) [reply ipn/ipnstate/ipnstate.go:30]: Added a doc line.

It's not linked to states at all. AuthURL is populated when control sends one. I believe we only send it right now when the client is in NeedsLogin, but we could explore other possible cases. E.g. maybe we want to send a pre-emptive reauth URL sometime in the future?

File: ipn/ipnlocal/local.go (hunk 2 above)
Comment (crawshaw) [reply ipn/ipnlocal/local.go:216]: As above, I don't want to start linking authURL to `state` given that it isn't anywhere else. If we decide to send more auth URLs for some reason, we shouldn't need to update the `tailscale status` code.

Comment (bradfitz) [reply ipn/ipnlocal/local.go:216]: So how's a user's supposed to use this?

You're presumably using this in Synology to do something, but when should a user go to this AuthURL. Documenting it as "oh, this is some field the control server sends sometimes.... meaning undefined, enjoy!" isn't particularly helpful.

But whatever. This isn't a stable API really. So submit away.

Comment (crawshaw) [reply ipn/ipnlocal/local.go:216]: My plan is to display it to the user whenever it's set along with a message "authenticate this client" or somesuch.

The problem is this is pretty undefined. I think current client behavior is to pop up an authURL in a browser under some (but not all?) conditions.

**Pull request merged**